lib/evaluate/expression.cc

```diff
@@ -471,7 +472,13 @@ template<int KIND>
 auto RealExpr<KIND>::IntPower::FoldScalar(FoldingContext &context,
```

Comment:
Why use `auto` for these functions? They don't seem to obey the "Don't use `auto` functions unless..." rule.

Reply:
This is an outlined member function definition whose result type uses the name of a type declared in the class.  Written without `auto`, the result type would have to be something like `std::optional<typename RealExpr<KIND>::Scalar>`.  This is exactly the situation in which the "unless..." exceptional language in the style guide is meant to apply.